src/QMCWaveFunctions/Fermion/MultiSlaterDeterminantFast.cpp

```diff
@@ -962,17 +962,27 @@ void MultiSlaterDeterminantFast::evaluateDerivatives(ParticleSet& P,
     }
   }
 
-  if (Dets.size() != 2)
-  {
-    throw std::runtime_error(
-        "MultiSlaterDeterminantFast::evaluateDerivatives only compatible with two quantum particle types.");
-  }
-  else
+  //Currently, the MultiDiracDeterminant::evaluateDerivatives works with a legacy design, essentially requiring only up and down determinants.
+  //e.g. for spinor cases, we only have one determinant so this interface doesn't work.
+  //Here we throw an error only if the optimization is turned on for MultiDiracDeterminants until the code is updated
+  bool all_Optimizable = true;
+  for (size_t id = 0; id < Dets.size() && all_Optimizable; id++)
+    all_Optimizable = Dets[id]->Optimizable;
+
+  if (all_Optimizable)
```

Comment:
This entire section could/should be refactored into a private method, its signature and name might even give offer up some clue as to what if any of the previous computation it has data dependence on and what it is.  I guess its the partial derivative with respect to the other spin multideterminant.

Reply:
added new private function evaluateMultiDiracDeterminantDerivatives to handle this